app/services/find_or_register_user.rb

```diff
@@ -0,0 +1,27 @@
+class FindOrRegisterUser
```

Comment:
I initially had this called `CreateUser` when I threw this together. I changed it to `RegisterUser` since Registration seems more in line with the user-facing behavior. `Create` to me is a specific operation on the database (it inserts a row in the users table). `Register` implies creation plus the other things we need to happen when a new user joins the platform (check SAM, maybe send a welcome email, etc). Still sort of agnostic here, but those were my reasons.